src/sentry/static/sentry/app/views/organizationEventsV2/relatedEvents.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+import {omit} from 'lodash';
+
+import {t} from 'app/locale';
+import SentryTypes from 'app/sentryTypes';
+import AsyncComponent from 'app/components/asyncComponent';
+import InlineSvg from 'app/components/inlineSvg';
+import Link from 'app/components/links/link';
+import Badge from 'app/components/badge';
+import Placeholder from 'app/components/placeholder';
+import ProjectBadge from 'app/components/idBadge/projectBadge';
+import DateTime from 'app/components/dateTime';
+import space from 'app/styles/space';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+
+import {MODAL_QUERY_KEYS} from './data';
+
+class RelatedEvents extends AsyncComponent {
+  static propTypes = {
+    event: SentryTypes.Event.isRequired,
+    location: PropTypes.object.isRequired,
+    organization: SentryTypes.Organization.isRequired,
+  };
+
+  getEndpoints() {
+    // TODO what happens when global-views feature is not on the org?
+    const {event, organization} = this.props;
+    const eventsUrl = `/organizations/${organization.slug}/events/`;
+    const trace = event.tags.find(tag => tag.key === 'trace');
+
+    if (!trace) {
+      return [];
+    }
+
+    const params = {
+      query: {
+        field: [
+          'project.name',
+          'title',
+          'transaction',
+          'id',
+          'issue.id',
+          'event.type',
+          'timestamp',
+        ],
+        sort: ['-timestamp'],
+        query: `trace:${trace.value}`,
+      },
+    };
+
+    return [['events', eventsUrl, params]];
+  }
+
+  renderLoading() {
+    return <Placeholder height="120px" bottomGutter={2} />;
+  }
+
+  renderBody() {
+    const {event, location} = this.props;
+    const {events} = this.state;
+    if (!events) {
+      return null;
+    }
+
+    return (
+      <Container>
+        <Title>
+          <InlineSvg src="icon-link" size="12px" /> {t('Related Events')}
+        </Title>
+        {events.length <= 1 ? (
+          <Card>{t('No related events found.')}</Card>
+        ) : (
+          events
+            // Don't show the current event. We don't have negation conditions
+            // in our API so we have to do this here.
+            .filter(item => item.id !== event.id)
+            .map(item => {
+              const eventUrl = {
+                pathname: location.pathname,
+                query: {
+                  ...omit(location.query, MODAL_QUERY_KEYS),
+                  eventSlug: `${item['project.name']}:${item.id}`,
+                },
+              };
+              const project = {slug: item['project.name']};
+              return (
+                <Card key={item.id}>
+                  <EventLink to={eventUrl} data-test-id="linked-event">
```

Comment:
Is `data-test-id` needed?

Reply:
I'm going to use it in an acceptance test.